ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -500,6 +500,90 @@ public void hardDeleteTest() throws InterruptedException, IOException, StoreExce
   }
 
   /**
+   * Tests that hard delete pause and resume where in hard deletes is done by the daemon thread and  not explicitly
+   * invoked by the tests
+   * @throws InterruptedException
+   * @throws IOException
+   * @throws StoreException
+   */
+  @Test
+  public void hardDeletePauseResumeTest() throws InterruptedException, IOException, StoreException {
+    properties.put("store.deleted.message.retention.days", Integer.toString(1));
+    properties.put("store.hard.delete.bytes.per.sec", Integer.toString(Integer.MAX_VALUE / 10));
+    properties.setProperty("store.data.flush.interval.seconds", "3600");
+    // enable daemon thread to run hard deletes
+    properties.put("store.enable.hard.delete", "true");
+    reloadIndex(false);
+    assertTrue("Hard delete is not running", index.hardDeleter.isRunning());
+    // IndexSegment still uses real time so advance time so that it goes 2 days past the real time.
+    advanceTime(SystemTime.getInstance().milliseconds() + 2 * Time.MsPerSec * Time.SecsPerDay);
+    // give it some time so that hard delete completes one cycle
+    Thread.sleep(50);
```

Comment:
How are you sure that the task you want completed completes within the sleep? This test looks like something that is liable to fail intermittently.

You have one of two options:-
1. Build a notification system into hard delete to notify on every cycle of hard delete (preferred - no one uses it now except tests but it is useful functionality anyway).
2. Sleep in small slices until the condition you want is met (i.e. StoreToken has reached the end). This is what the integration test does.